src/pages/[platform]/build-a-backend/functions/examples/create_user_profile_record/index.mdx

```diff
@@ -0,0 +1,123 @@
+import { getCustomStaticPath } from '@/utils/getCustomStaticPath';
+
+export const meta = {
+  title: 'Create a user profile record',
+  description:
+    'Use an Auth Post Authentication trigger to automatically a user profile record',
+  platforms: [
+    'android',
+    'angular',
+    'flutter',
+    'javascript',
+    'nextjs',
+    'react',
+    'react-native',
+    'swift',
+    'vue'
+  ]
+};
+
+export function getStaticPaths() {
+  return getCustomStaticPath(meta.platforms);
+}
+
+export function getStaticProps() {
+  return {
+    props: {
+      meta
+    }
+  };
+}
+
+<Callout warning>
+
+**Under active development:** This use case is not currently supported with social provider sign-in. The experience may change between versions of `@aws-amplify/backend`. Try it out and provide feedback at https://github.com/aws-amplify/amplify-backend/issues/new/choose
+
+</Callout>
+
+You can use `defineAuth` and `defineFunction` to create a [Cognito post confirmation Lambda trigger](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-post-confirmation.html) to create a profile record when a user is confirmed.
+
+<Callout info>
+
+A user is "confirmed" when they verify their account. Typically this happens when the user confirms their email via the verification email. The post confirmation handler will _not_ be triggered for federated sign-ins (i.e. social sign-in).
+
+</Callout>
+
+To get started, Update the `amplify/data/resource.ts` file to define a data model for the user's profile:
+
+```ts title="amplify/data/resource.ts"
+import { type ClientSchema, a, defineData } from "@aws-amplify/backend";
+
+const schema = a.schema({
+  Profile: a.model({
+    name: a.string(),
+    email: a.string(),
+  }),
+});
+
+export type Schema = ClientSchema<typeof schema>;
+
+export const data = defineData({
+  schema,
+  authorizationModes: {
+    defaultAuthorizationMode: "apiKey",
+    // API Key is used for a.allow.public() rules
+    apiKeyAuthorizationMode: {
+      expiresInDays: 30,
+    },
+  },
+});
+
+```
+
+
+Next, create a new directory and a resource file, `amplify/auth/post-confirmation/resource.ts`. Then, define the Function with `defineFunction`:
+
+```ts title="amplify/auth/post-confirmation/resource.ts"
+import { defineFunction } from '@aws-amplify/backend';
+
+export const postConfirmation = defineFunction({
+  entry: './post-confirmation.ts',
+});
+```
+
+Then, create the corresponding handler file, `amplify/auth/post-confirmation/handler.ts`, file with the following contents:
+
+```ts title="amplify/auth/post-confirmation/handler.ts"
+import type { PostConfirmationTriggerHandler } from "aws-lambda";
+import { generateClient } from "aws-amplify/api";
+import { type Schema } from "../../data/resource";
+
```

Comment:
there are a few things missing here because the `Amplify.configure()` hasn't happened and the resource authorization rule to create records.

Reply:
Updated